### PR TITLE
feat(meeting): initialize recording property from the locus info

### PR DESCRIPTION
**Pull request by akoushke**
Fixes #[INSERT LINK TO ISSUE NUMBER]
SPARK-178087
---
currently we don't which meetings are supporting the `recording` feature by locus. this check will allow us to identify that. if `recording` is null, it means that the meeting does not support recording AKA 1:1 and free trial meetings.

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 1]
```diff
@@ -619,7 +619,7 @@ export default class Meeting extends StatelessWebexPlugin {
      * @public
      * @memberof Meeting
      */
-    this.recording = null;
+    this.recording = attrs.locus?.controls.record ? {state: RECORDING_STATE.IDLE, modifiedBy: '', lastModified: ''} : null;
```
Comment (akoushke) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:622]: I do prefer to define classes for different controls with an abstract called `Control` and have classes `Recording` `Locked` etc etc.

**Review by arun3528 — CHANGES_REQUESTED** [review]
This logic needs to be moved in the locusInfo updateControl section , Placing it here wont trigger an event to the developer initially on recording ,

**Comment by akoushke** [comment]
> This logic needs to be moved in the locusInfo updateControl section , Placing it here wont trigger an event to the developer initially on recording ,

and it shouldn't trigger an event at the initial level IMO.  events get triggered after the setup and `recording` should be defined before that the event setup.

**Comment by akoushke** [comment]
I tested this and worked as expected.

**Comment by akoushke** [comment]
1:1 
Screen Shot 2020-09-22 at 6 27 35 PM
meeting
Screen Shot 2020-09-22 at 6 28 26 PM

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/controlsUtils.js [hunk 2]
```diff
@@ -41,19 +41,16 @@ ControlsUtils.parse = (controls) => {
...
-        hasRecord(previous) && previous.record.paused,
-        current.record.paused
+      hasRecordingPausedChanged: current?.record && !isEqual(
+        previous?.record?.paused, current.record.paused
       ),
-      hasRecordingChanged: hasRecord(current) && !isEqual(
-        hasRecord(previous) && previous.record.recording,
-        current.record.recording
+      hasRecordingChanged: current?.record && !isEqual(
+        previous?.record?.recording, current?.record?.recording
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/controlsUtils.js:53]: nice so this will check undefined with false

**Review by arun3528 — APPROVED** [review]

**Pull request merged**